### PR TITLE
Add WPS Office

### PR DESCRIPTION
**Pull request by gasinvein**
WPS Office is a full-featured proprietary office suite.
Supersedes #309

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.wps.Office
testbot, build com.wps.Office

**Comment by LeandroStanger** [comment]
```
Error: Failed to install com.wps.Office/x86_64/test: GDBus.Error:org.gtk.GDBus.UnmappedGError.Quark._as_2dstore_2derror_2dquark.Code0: Failed to parse /var/lib/flatpak/appstream/com.wps.Office-origin/x86_64/active/appstream.xml.gz file: Error getting information for file “/var/lib/flatpak/appstream/com.wps.Office-origin/x86_64/active/appstream.xml.gz”: File or directory not found
```

**Comment by aiamuzz** [comment]
> ```
> Error: Failed to install com.wps.Office/x86_64/test: GDBus.Error:org.gtk.GDBus.UnmappedGError.Quark._as_2dstore_2derror_2dquark.Code0: Failed to parse /var/lib/flatpak/appstream/com.wps.Office-origin/x86_64/active/appstream.xml.gz file: Error getting information for file “/var/lib/flatpak/appstream/com.wps.Office-origin/x86_64/active/appstream.xml.gz”: File or directory not found
> ```

really ? ... interesting ... i was able to install and run it ...

**Comment by gasinvein** [comment]
@LeandroStanger  I didn't edit the appdata, except for changing IDs. Did the builds from previous PR work for you?

**Comment by LeandroStanger** [comment]
@gasinvein I did not test the previous PR

File: com.wps.Office.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        path: wps-i18n-makefile-destdir.patch
+
+  - name: wps
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 apply_extra.sh /app/bin/apply_extra
+      - ln -s ../extra/wps-office/wps /app/bin/wps
+      - ln -s ../extra/wps-office/et /app/bin/et
+      - ln -s ../extra/wps-office/wpp /app/bin/wpp
+      - install -Dm644 ${FLATPAK_ID}.appdata.xml -t /app/share/appdata/
```
[the remark this answers is not in the document]
Comment (gasinvein) [reply com.wps.Office.yml]: Appdata is installed just fine, and the build bot [checks it](http://repo-test.flathub.org:8010/#/builders/12/builds/346).

**Comment by nedrichards** [comment]
Installs fine for me. I get the 'missing symbol fonts' error at startup that links to http://community.wps.cn/wiki/No_necessary_symbol_fonts I also get `QGtkStyle could not resolve GTK. Make sure you have installed the proper libraries.` but I don't think that's a big issue.

Endless OS doesn't particularly like showing the base com.wps.Office on it's main app grid but that's not your problem, more one for @ramcq other than that, lgtm.

**Comment by gasinvein** [comment]
@nedrichards Seems like those fonts are actually Microsoft web core fonts. The are available from some shady sources, but obviously non-free and I'm not sure if we should package them.

**Comment by barthalion** [comment]
LGTM

**Comment by nedrichards** [comment]
Yeah, probably fair.

File: com.wps.Office.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    <content_attribute id="social-info">none</content_attribute>
+    <content_attribute id="social-audio">none</content_attribute>
+    <content_attribute id="social-location">none</content_attribute>
+    <content_attribute id="social-contacts">none</content_attribute>
+    <content_attribute id="money-purchasing">none</content_attribute>
+    <content_attribute id="money-gambling">none</content_attribute>
+  </content_rating>
+</component>
+<component type="desktop-application">
+  <id>com.wps.Office.et</id>
```
Comment (gasinvein) [inline com.wps.Office.appdata.xml]: appdata items other than `com.wps.Office` aren't exposed in app stores. Maybe they should be removed?

Comment (nedrichards) [reply com.wps.Office.appdata.xml]: No, we want this - it's the same for LibreOffice. There are plans to have all this exposed within app stores (certainly GNOME Software) and we need the metadata there first (same as OARS).

Comment (ramcq) [reply com.wps.Office.appdata.xml]: Right, they are not different Flatpaks, but they are different launchables. Flatpak allows exporting any "sub-IDs" in the appstream, although it's gently broken in the LibreOffice Flatpak at the moment.

**Comment by gasinvein** [comment]
@amtlib-dot-dll @cheese do you want to help maintaining it?

**Comment by cheese** [comment]
> @amtlib-dot-dll @cheese do you want to help maintaining it?

Yes. I am grad to help.

**Pull request closed without merging**